runbotics/runbotics-scheduler/src/scheduler-database/tags/tag.controller.ts

```diff
@@ -0,0 +1,64 @@
+import { BadRequestException, Body, Controller, Delete, Get, HttpCode, HttpStatus, Param, ParseIntPipe, ParseUUIDPipe, Post, Query, UseInterceptors } from '@nestjs/common';
+
+import { TenantInterceptor } from '#/utils/interceptors/tenant.interceptor';
+import { Logger } from '#/utils/logger';
+import { User } from '#/utils/decorators/user.decorator';
+import { UserEntity } from '#/database/user/user.entity';
+
+import { TagService } from './tag.service';
+import { Tenant } from 'runbotics-common';
+import { ZodValidationPipe } from '#/utils/pipes/zod-validation.pipe';
+import { CreateTagDto, createTagSchema } from './dto/create-tag.dto';
+
+@UseInterceptors(TenantInterceptor)
+@Controller('/api/scheduler/tenants/:tenantId/tags')
+export class TagController {
+    private readonly logger = new Logger(TagController.name);
+
+    constructor(private readonly tagService: TagService) {}
+    
+    @Get()
+    async getAllTags(
+        @Param('tenantId', ParseUUIDPipe) tenantId: Tenant['id'],
+        @Query('name.contains') searchPhrase: string | undefined
+    ) {
+        this.logger.log(`REST request to get all tags for the tenant: ${tenantId}`);
+        return await this.tagService.getAll(tenantId, searchPhrase);
```

Comment:
async/await is redundant here

Reply:
also I would set `searchPhrase` as optional rather than using `undefined`

Reply:
How can I make query optional? was looking for the solution, but couldn't find any so I decided to use undefined instead.

Reply:
I had something like this in mind

`@Query('name.contains') searchPhrase?: string`

Reply:
that doesn't work here in controller where there is a Query decorator